File: Marathon/findParticipant.js
function solution(participant, completion) {
    var answer = "";

    var participantCount = participant.reduce(countName, {}); // 初期値は{}
    var completionCount = completion.reduce(countName, {}); // 初期値は{}

    // Object.keys(): オブジェクトのキーが格納された新規配列を作成
    // そして新規作成された配列に対してforEach()でループするので、エラーにならない
    Object.keys(participantCount).forEach(function(key) {
        if (typeof completionCount[key] === "undefined" || participantCount[key] !== completionCount[key]) {
            answer = key;
        }
    });
    return answer;
}

/**
 * 配列内の重複した要素を数える
 * 
 * @param {*} prev 
 * @param {*} current 
 * return prev
 */
function countName(prev, current) {
    prev[current] = (prev[current] || 0) + 1;
    return prev;
}

solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]);

File: Marathon/Find-Participant.py
# マラソンで完走出来なかった選手探し.
#
# 【条件】
# 完走できなかった選手は1人のみ.
# 同姓同名あり.
def solution(participant, completion):
    # マラソンで完走出来なかった選手
    answer = ''
    # sort(): 元のリストをソート
    # sorted(): ソートした新たなリストを生成
    sorted_participant = sorted(participant)
    sorted_completion = sorted(completion)

    for index in range(len(sorted_completion)):
        if sorted_participant[index] != sorted_completion[index]:
            answer = sorted_participant[index]
            return answer

    return answer

def main():
    print(solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"]))

if __name__ == '__main__':
    main()